Update README with direnv+asdf python workaround (#303)
elixir-nx/nx: exla/README.md

#### Python and asdf

> `Bazel` cannot find `python` installed via the `asdf` version manager by default.`asdf` uses a function to lookup the specified version of a given binary, this approach prevents `Bazel` from being able to correctly build `EXLA`. The error is `unknown command: python. Perhaps you have to reshim?`. There are two known workarounds:

Reviewer: ```suggestion
`Bazel` cannot find `python` installed via the `asdf` version manager by default. `asdf` uses a function to lookup the specified version of a given binary, this approach prevents `Bazel` from being able to correctly build `EXLA`. The error is `unknown command: python. Perhaps you have to reshim?`. There are two known workarounds:
```

> Note the build process looks for `python`, not `python3`. You may need to remove `~/.cache/exla` to clean up from previous build attempts.
> 
> 
> 2. Use the [`asdf direnv`](https://github.com/asdf-community/asdf-direnv) plugin to install [`direnv 2.20.0`](https://direnv.net). `direnv` along with the `asdf-direnv` plugin will explicitly set the paths for any binary specified in your project's `.tool-version` files.
> 
> ```
> asdf plugin-add direnv
> asdf install direnv 2.20.0
> asdf global  direnv 2.20.0
> ```
> 
> Add the `asdf direnv hook` to your shell profile:
> 
> **BASH** Add the following line to the bottom of your `~/.bashrc`
> ```
> eval "$(direnv hook bash)"
> direnv() { asdf exec direnv "$@"; }
> ```
> 
> **ZSH** Add the following line to the bottom of your `~/.zshrc`
> ```
> eval "$(direnv hook zsh)"
> direnv() { asdf exec direnv "$@"; }
> ```
> 
> Next, add the `asdf-direnv` plugin to the `direnvrc` configuration file `~/.config/direnv/direnvrc`:
> 
> ```
> source "$(asdf direnv hook asdf)"
> ```
> Then, in the root of your `Nx` project add an `.envrc` file with the following: `use asdf`.
> 
> `direnv` will now prompt you to allow the changes to the `.envrc` file, you can explicitly allow it with: `direnv allow`. You should now see `direnv` output a list of which `asdf` plugins it is going to set paths for, and any environment variables you choose to set using it. If your `.tool-versions` are all correctly set, `exla` should now be able to successfully compile.
> 
> Note after setting up `direnv` it is high recommended you clear the build cache by removing ` ~/.cache/exla`.

Reviewer: Those instructions look precise but they are too specific to direnv. It is probably best if we just point towards that direction and let users set it up.

```suggestion
Note the build process looks for `python`, not `python3`.

2. Use the [`asdf direnv`](https://github.com/asdf-community/asdf-direnv) plugin to install [`direnv 2.20.0`](https://direnv.net). `direnv` along with the `asdf-direnv` plugin will explicitly set the paths for any binary specified in your project's `.tool-version` files.

After doing any of the steps above, it may be necessary to clear the build cache by removing ` ~/.cache/exla`.
```
